message.go

```diff
@@ -471,6 +471,16 @@ func (r RedisResult) ToAny() (v any, err error) {
 	return
 }
 
+// IsMap delegates to RedisMessage.IsMap
+func (r RedisResult) IsMap() bool {
+	return r.val.IsMap()
+}
+
+// IsArray delegates to RedisMessage.IsArray
+func (r RedisResult) IsArray() bool {
+	return r.val.IsArray()
+}
```

Comment:
```suggestion
func (r RedisResult) isMap() bool {
	return r.val.IsMap()
}

// IsArray delegates to RedisMessage.IsArray
func (r RedisResult) isArray() bool {
	return r.val.IsArray()
}
```

Can we make these 2 methods private first? I am concerned about making these direct delegations to `r.val` as public APIs. I think we should at least ask users to check the `r.err` first.

Reply:
But if we make `IsMap` private, functions in `command.go` can not access it,
I think we could check if any error exists, then delegate to `r.val.IsMap`.

```
func (r RedisResult) IsMap() bool {
	if r.err != nil {
		panic(fmt.Errorf("RedisResult has error: %w, please check the error before calling RedisResult.IsMap", r.err))
	}
	return r.val.IsMap()
}
```

Reply:
UPDATE: I'll use `res.ToMessage()` to get underlying message first, then do the `IsMap`, `IsArray` check.